modules/couchbase/src/main/java/org/testcontainers/containers/CouchbaseContainer.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2016 Couchbase, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.testcontainers.containers;
+
+import com.couchbase.client.core.message.config.RestApiResponse;
+import com.couchbase.client.core.utils.Base64;
+import com.couchbase.client.deps.io.netty.handler.codec.http.HttpHeaders;
+import com.couchbase.client.java.CouchbaseCluster;
+import com.couchbase.client.java.bucket.BucketType;
+import com.couchbase.client.java.cluster.BucketSettings;
+import com.couchbase.client.java.cluster.DefaultBucketSettings;
+import com.couchbase.client.java.cluster.api.ClusterApiClient;
+import com.couchbase.client.java.cluster.api.RestBuilder;
+import com.couchbase.client.java.document.json.JsonArray;
+import com.couchbase.client.java.env.CouchbaseEnvironment;
+import com.couchbase.client.java.env.DefaultCouchbaseEnvironment;
+import com.couchbase.client.java.query.Index;
+import com.couchbase.client.java.query.N1qlQuery;
+import lombok.Cleanup;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.traits.LinkableContainer;
+import org.testcontainers.containers.wait.HttpWaitStrategy;
+import org.testcontainers.shaded.com.github.dockerjava.api.command.InspectContainerResponse;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Laurent Doguin
+ */
+public class CouchbaseContainer<SELF extends CouchbaseContainer<SELF>> extends GenericContainer<SELF> {
+
+    private String memoryQuota = "400";
+
+    private String indexMemoryQuota = "400";
+
+    private String clusterUsername = "Administrator";
+
+    private String clusterPassword = "password";
+
+    private Boolean keyValue = true;
+
+    private Boolean query = true;
+
+    private Boolean index = true;
+
+    private Boolean fts = true;
+
+    private Boolean beerSample = false;
+
+    private Boolean travelSample = false;
+
+    private Boolean gamesIMSample = false;
+
+    private CouchbaseEnvironment couchbaseEnvironment;
+
+    private CouchbaseCluster couchbaseCluster;
+
+    private List<BucketSettings> newBuckets = new ArrayList<>();
+
+    private String urlBase;
+
+    public CouchbaseContainer() {
+        super("couchbase/server:latest");
```

Comment:
could you please lock the version? We try to avoid ":latest" everywhere. Thanks!